skbio/stats/power.py

```diff
@@ -870,49 +866,276 @@ def _compare_distributions(test, samples, counts=5, mode="ind", num_iter=1000):
 
     """
 
-    # Determines the number of groups
-    num_groups = len(samples)
+    # Prealocates the pvalue matrix
+    p_values = np.zeros((num_p, num_iter))
 
-    # Checks the mode
-    if mode not in {'ind', 'matched'}:
-        raise ValueError('Supported sample modes are "ind" and "matched".')
+    # Determines the number of samples per group
+    num_groups = len(samples)
+    samp_lens = [len(sample) for sample in samples]
 
-    # Handles the number of samples for later instances
     if isinstance(counts, int):
         counts = np.array([counts] * num_groups)
 
-    if not len(counts) == num_groups:
-        raise ValueError('If counts is a 1-D array, there must be a count to'
-                         ' draw for each group.')
-
-    # Checks the group length
-    samp_lens = [len(sample) for sample in samples]
-    # Checks the group length
-    if mode == 'matched' and np.array([samp_lens[i] != samp_lens[i+1] for i in
-                                       range(num_groups-1)]).all():
-        raise ValueError('In "matched" mode, each sample must have the same'
-                         ' number of observations.')
-    if np.array([samp_lens[i] < counts[i] for i in range(num_groups)]).any():
-        raise ValueError('You cannot choose more observations that exist '
-                         'in a sample.')
-
-    # Prealocates the pvalue matrix
-    p_values = np.zeros((num_iter))
-
     for idx in range(num_iter):
         if mode == "matched":
             pos = np.random.choice(np.arange(0, samp_lens[0]), counts[0],
                                    replace=False)
             subs = [sample[pos] for sample in samples]
+        elif mode == "paired":
+            subs = _draw_paired_samples(*samples)
         else:
             subs = [np.random.choice(np.array(pop), counts[i], replace=False)
                     for i, pop in enumerate(samples)]
 
-        p_values[idx] = test(subs)
+        p_values[:, idx] = test(subs)
+
+    if num_p == 1:
```

Comment:
I think you can get rid of this statement - its a little redundant.  `squeeze()` won't do anything, unless `num_p==1`